core/shared/src/test/scala/zio/RefSpec.scala

```diff
@@ -96,4 +118,71 @@ class RefSpec extends Specification with DefaultRuntime {
         value <- ref.modifySome[String]("State doesn't change") { case Closed => ("active", Active) }
       } yield value must beTheSameAs("State doesn't change")
     )
+
+  def e9 =
+    unsafeRun(
+      for {
+        ref    <- Ref.make(Active)
+        refInt = ref.map(toInt)
+        value  <- refInt.get.get
+      } yield value must_=== 1
+    )
+
+  def e10 =
+    unsafeRun(
+      for {
+        ref <- Ref.make[State](Active)
+        refInt = ref.contramap { i: Int =>
+          fromInt(i).get
+        }
+        _     <- refInt.set(2)
+        value <- refInt.get
+      } yield value must_=== Changed
+    )
+
+  def e11 = {
+    val failure = new Error("expected")
+    unsafeRun(
+      for {
+        ref    <- Ref.make[Option[State]](Some(Active))
+        refEA  = ref.mapError(_ => "oh no !")
+        failed <- refEA.set(Some(Changed)).map(_ => Option.empty[State]).someOrFail(failure).either
```

Comment:
This test is wrong, I am not sure how to test `mapError`, need some help here @jdegoes

Reply:
To test `mapError`, you'll have to make a `Ref` whose getter will fail. You can do that manually (`new ZRef { def get = ZIO.fail("foo") ... }`) or through another "lifting" operation that we have yet to add. Then you can map that error into another one, and test the old and new error.

Let me know which approach you'd prefer!

Reply:
cool I went with the manual way for now, not sure how this `lift` function would look, if it would make this cleaner I am game